kubernetes/utils/create_from_yaml.py

```diff
@@ -14,60 +14,119 @@
+    Perform an action from a yaml file.
+
+    :param k8s_client: an ApiClient object, initialized with the client args
+    :yaml_path: the path on the file system to the yaml file
+    :type yaml_path: str
+    :param verbose: print confimation information
+    :type verbose: bool
+
+    Available parameters for performing the subsequent action:
+    :param async_req:
+    :type async_req: bool
+    :param include_uninitialized: include partially initialized resources in
+    the response.
+    :type include_uninitialized: bool
+    :param str pretty: pretty print the output
+    :param str dry_run: When present, indicates that modifications should not
+    be persisted. An invalid or unrecognized dry_run directive will result in
+    an error response and no further processing of the request.
+    Valid values are: - All: all dry run stages will be processed
+    """
+    with open(path.abspath(yaml_path)) as f:
+        k8s_api = create_from_yaml(k8s_client, f.read(), verbose=verbose,
+                                   **kwargs)
+
+    return k8s_api
+
+
+def create_from_yaml(k8s_client, yaml_str, verbose=False, **kwargs):
+    """
+    Perform an action from a yaml string.
+
+    :param k8s_client: an ApiClient object, initialized with the client args
+    :yaml_str: a string containing valid yaml or json.
+    :type yaml: str
+    :param verbose: print confimation information
+    :type verbose: bool
 
     Available parameters for performing the subsequent action:
-    :param async_req bool
-    :param bool include_uninitialized: If true, partially initialized resources are included in the response.
-    :param str pretty: If 'true', then the output is pretty printed.
-    :param str dry_run: When present, indicates that modifications should not be persisted. An invalid or unrecognized dryRun directive will result in an error response and no further processing of the request. Valid values are: - All: all dry run stages will be processed
+    :param async_req:
+    :type async_req: bool
+    :param include_uninitialized: include partially initialized resources in
+    the response.
+    :type include_uninitialized: bool
+    :param str pretty: pretty print the output
+    :param str dry_run: When present, indicates that modifications should not
+    be persisted. An invalid or unrecognized dry_run directive will result in
+    an error response and no further processing of the request.
+    Valid values are: - All: all dry run stages will be processed
+    """
+    stream = yaml.safe_load_all(yaml_str)
+    for obj in stream:
+        k8s_api = create_from_map(k8s_client, obj, verbose, **kwargs)
+
+    return k8s_api
+
+
+def create_from_map(k8s_client, yml_object, verbose, **kwargs):
     """
+    performe an action from a valid parsed yaml object (as dict).
+
+    :param k8s_client: an ApiClient objcet initialized with the client args
+    :yml_object dict: a parsed yaml object
```

Comment:
I don't think one needs to explicitly reference yaml here as what the function wants is a kubernetes manifest in python dictionary form
though it probably also does not harm as yaml is the most common representation form of that data.

Reply:
this is why I added `create_from_map`